File: Anul II/Medii de proiectare si programare/Laborator/lab06/services/src/main/java/com/ubb/IMainObserver.java
package com.ubb;

import com.ubb.exceptions.ContestDataException;
import com.ubb.model.Participant;
import com.ubb.dto.RaceDTO;

import java.util.List;

public interface IMainObserver {
    void raceAdded(RaceDTO race) throws ContestDataException;
    void participantAdded(Participant participant) throws ContestDataException;
    void raceEntriesAdded(List<RaceDTO> races) throws ContestDataException;
}

File: Anul II/Medii de proiectare si programare/Laborator/lab07/Ubb.BikeContest.Service/IMainObserver.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Ubb.BikeContest.Model;
using Ubb.BikeContest.Model.data;

namespace Ubb.BikeContest.Services
{
    public interface IMainObserver
    {
        void ParticipantAdded(Participant participant);
        void RaceEntriesAdded(IEnumerable<RaceDto> races);
    }
}
